### ISSUE TITLE
Slow fallback implementation invoked in DCGAN MNIST example

### ISSUE THREAD

**Issue by Dsantra92** [FluxML/model-zoo#318]
The [dcgan_mnist tutorial](https://github.com/FluxML/model-zoo/blob/master/vision/dcgan_mnist/dcgan_mnist.jl) throws warning:

```
┌ Warning: Slow fallback implementation invoked for ∇conv_data!  You probably don't want this; check your datatypes.
│   yT = Float32
│   T1 = Float32
│   T2 = Float64
└ @ NNlib ~/.julia/packages/NNlib/P9BhZ/src/conv.jl:291
``` 
This is beacuse the weights for the ConvTranspose layers are accidentally set to Float64 due to the weight init function. This can be simply fixed by using 0.02f0(Float32) instead of 0.02(Float64). To be precise, the [dcgan_init](https://github.com/FluxML/model-zoo/blob/d1a598a5f5b634f765aa58c4d963925e77f05264/vision/dcgan_mnist/dcgan_mnist.jl#L35) function should be,
```julia
dcgan_init(shape...) = randn(Float32, shape...) * 0.02f0
```

**Comment by ToucheSir** [comment]
Good catch, do you mind filing a quick PR so we can give you the honours for fixing this?

**Comment by Dsantra92** [comment]
Looking back at it, there might some other cases in the tutorial where there can be a potential conflict of Float32 and Float64. I will make a pr within today to fix those too.
